R/io.R

```diff
@@ -356,24 +361,48 @@ write_civis.numeric <- function(x, tablename, database = NULL, if_exists = "fail
 #' plan(multisession)
 #' file_id <- write_civis_file("my_large_file")
 #' }
-#' @details By default, R objects are serialized using \code{\link{saveRDS}} before uploading the object
+#' @details
+#' Data frames are uploaded as CSVs using \code{\link{write.csv}}, with row.names = FALSE
+#' by default. Additional arguments to \code{write.csv} can be passed through \code{...}.
+#'
+#' By default, R objects are serialized using \code{\link{saveRDS}} before uploading the object
 #' to the files endpoint. If given a filepath, the file is uploaded as-is.
+#'
+#'
 #' @family io
 #' @export
 write_civis_file <- function(x, ...) {
   UseMethod("write_civis_file")
 }
 
 #' @export
-#' @describeIn write_civis_file Serialize R object to Civis Platform (Files endpoint).
+#' @describeIn write_civis_file Serialize R object
 write_civis_file.default <- function(x, name = 'r-object.rds', expires_at = NULL, ...) {
   with_tempfile(function(tmp_file, ...) {
     saveRDS(x, file = tmp_file, ...)
     write_civis_file.character(x = tmp_file, name = name, expires_at = expires_at)
   })
 }
 
-#' @describeIn write_civis_file Upload a text file to Civis Platform (Files endpoint).
+#' @describeIn write_civis_file Upload a data frame as a csv
+#' @param row.names default FALSE. Either a logical value indicating whether the row names
+#' of x are to be written along with x, or a character vector of row names to be written.
+#' @export
+#' @importFrom utils write.csv
+write_civis_file.data.frame <- function(x,
+                                        name = 'data.csv',
+                                        expires_at = NULL,
+                                        row.names = FALSE,
+                                        ...) {
+  with_tempfile(function(tmp_file, ...) {
+    write.csv(x, file = tmp_file, row.names = row.names, ...)
+    write_civis_file.character(x = tmp_file,
+                               name = name,
+                               expires_at = expires_at)
+  })
+}
```

Comment:
if I understand `with_tempfile` correctly, should this read:

```R
  with_tempfile(function(tmp_file, ...) {
    write.csv(x, file = tmp_file, row.names = row.names, ...)
    write_civis_file.character(x = tmp_file,
                               name = name,
                               expires_at = expires_at)
  }, name=name, expires_at=expires_at, row.names=row.name, ...)
```
?

Maybe this works as is.

Reply:
It does work as-is, because those args are stored in the enclosing environment of the temporary function. Too tricky?

Reply:
It looks pretty clear to me, just wanted to double check.